lib/ht_sip_validator/validator/checksums/expected_value.rb

```diff
@@ -6,26 +6,39 @@
 module HathiTrust::Validator::Checksums
   # validates that checksums file values match calculated values
   class ExpectedValue < HathiTrust::Validator::Base
+    attr :calculated
 
     def perform_validation
       checksumed_files = @sip.files - EXEMPT_FILENAMES
+      @calculated = calculate_checksums(checksumed_files)
+
       checksumed_files.map do |filename|
         if @sip.checksums.checksum_for(filename) != calculated_checksum_for(filename)
-          create_error(
-            validation_type: :expected_checksum_value,
-            human_message: "Checksum mismatch for #{filename}",
-            extras: { filename: filename }
-          )
+          error_for filename
         end
       end
     end
 
-    # WARN! using private method
-    def calculated_checksum_for(file)
-      @sip.send(:file_in_zip, file) do | filestream |
-        Digest::MD5.hexdigest(filestream.read)
-      end
+    def calculated_checksum_for(filename)
+      @calculated[filename]
+    end
+
+    def error_for(filename)
+      create_error(
+        validation_type: :expected_checksum_value,
+        human_message: "Checksum mismatch for #{filename}",
+        extras: { filename: filename }
+      )
     end
 
+    private 
+
+    def calculate_checksums(filenames)
```

Comment:
Looks like the intention is to not bother checking the files in `filenames`, but that isn't the actual behavior